src/BinSkim.Driver/RoslynAnalyzer/ILOperations.cs

```diff
@@ -0,0 +1,485 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.CodeAnalysis.Semantics;
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.IL
+{
+    // Things that are needed from a Compilation to ensure various invariants.
+    internal interface ISymbolProvider
+    {
+        ITypeSymbol CreatePointerType(ITypeSymbol pointedAtType);
+
+        // FEEDBACK: Do we need to support custom bounds? (Not supported in VB/C#)
+        ITypeSymbol CreateArrayType(ITypeSymbol elementType, int rank);
+
+        ITypeSymbol GetSpecialType(SpecialType specialType);
+    }
+
+    internal abstract class Operation : IOperation
+    {
+        public abstract OperationKind Kind { get; }
+
+        // TODO: Fake this out. 
+        // FEEDBACK: Should this be optional, with separate location?
+        SyntaxNode IOperation.Syntax => null;
```

Comment:
Yes. I'm not even sure that having the Syntax property is a good idea, given that its primary purpose is to provide location information.